src/ls_server.rs

```diff
@@ -251,34 +252,36 @@ impl LsService {
     }
 
     fn init(&self, id: usize, init: InitializeParams) {
-        let result = InitializeCapabilities {
+        let result = InitializeResult {
             capabilities: ServerCapabilities {
-                textDocumentSync: DocumentSyncKind::Incremental as usize,
-                hoverProvider: true,
-                completionProvider: CompletionOptions {
-                    resolveProvider: true,
-                    triggerCharacters: vec![".".to_string()],
-                },
+                text_document_sync: Some(TextDocumentSyncKind::Incremental),
```

Comment:
I guess it depends on what you are modelling - if there is a case for a non-null Foo, then I would agree, if the Foo intrinsically has a null case then I think it is better to reflect that in the enum. Does the spec always allow null for these enums? If so I'd argue that using Option<Foo> rather than extending Foo is unnecessary complexity (YAGNI). If there are places in the spec where you are allowed non-null, then keep them as is.